config-tab.js

```diff
@@ -183,54 +211,42 @@ export async function initializeConfigTab() {
       `;
     }
 
-    if (!url) {
-      testStatus.innerHTML = `
-        <span class="status-indicator status-error"></span>
-        ⌨️ Please enter a URL to test
+    if (shouldArchive) {
+      // Show loading state
+      testButton.disabled = true;
+      testStatus.innerHTML += `
+        <span id="inprogress-test">
+          &nbsp; &nbsp; <span class="spinner-border spinner-border-sm" role="status" aria-hidden="true"></span>
+          Submitting...
+        </span>
       `;
-      return;
-    }
 
-    // Show loading state
-    testButton.disabled = true;
-    testStatus.innerHTML += `
-      <span id="inprogress-test">
-        &nbsp; &nbsp; <span class="spinner-border spinner-border-sm" role="status" aria-hidden="true"></span>
-        Submitting...
-      </span>
-    `;
+      try {
+        const testSnapshot = {
```

Comment:
shouldn't this use the same `const testSnapshot = new Snapshot(...)` type that we use everywhere else?